e2e/puppeteer-custom-environment.ts

```diff
@@ -18,28 +15,28 @@ class PuppeteerCustomEnvironment extends PuppeteerEnvironment {
   }
 
   // Take a screenshot right after failure
-  async handleTestEvent(event, state) {
+  async handleTestEvent(event) {
     switch (event.name) {
-    case 'test_fn_failure':
-      if (state.currentlyRunningTest.invocations > retryTimes) {
-        console.error(`Failed test:  "${event.test.name}"`);
-        const testName = fp.startCase(state.currentlyRunningTest.name).replace(/[^A-Z0-9]+/ig, '');
-        const screenshotDir = 'logs/screenshot';
-        await fs.ensureDir(screenshotDir);
+      // case 'test_fn_failure':
+      // case 'hook_failure':
+      case 'test_done':
```

Comment:
Case `test_fn_failure` were not working in some cases of failures such as unhandled exceptions, thus no screenshot were taken on failure. Change to case `test_done` and `event.test.errors.length > 0` to cover all types of failures